roles/etcd/tasks/install_etcdctl_docker.yml

```diff
@@ -1,11 +1,36 @@
 ---
-- name: Install | Copy etcdctl binary from docker container
-  command: sh -c "{{ docker_bin_dir }}/docker rm -f etcdctl-binarycopy;
-           {{ docker_bin_dir }}/docker create --name etcdctl-binarycopy {{ etcd_image_repo }}:{{ etcd_image_tag }} &&
-           {{ docker_bin_dir }}/docker cp etcdctl-binarycopy:/usr/local/bin/etcdctl {{ bin_dir }}/etcdctl &&
-           {{ docker_bin_dir }}/docker rm -f etcdctl-binarycopy"
-  register: etcdctl_install_result
-  until: etcdctl_install_result.rc == 0
-  retries: "{{ etcd_retries }}"
+- name: Set commands for etcdctl container tasks
+  set_fact:
+    etcdctl_compare_command: >-
+      {%- if container_manager in ['docker', 'crio'] %}
+      {{ docker_bin_dir }}/docker run --rm -v {{ bin_dir }}:/systembindir --entrypoint /usr/bin/cmp {{ etcd_image_repo }}:{{ etcd_image_tag }} /usr/local/bin/etcdctl /systembindir/helm
+      {%- elif container_manager == "containerd" %}
+      ctr run --rm --mount type=bind,src={{ bin_dir }},dst=/systembindir,options=rbind:rw {{ etcd_image_repo }}:{{ etcd_image_tag }} etcdctl-compare sh -c 'cmp /usr/local/bin/etcdctl /systembindir/etcdctl'
+      {%- endif %}
+    etcdctl_copy_command: >-
+      {%- if container_manager in ['docker', 'crio'] %}
```

Comment:
Are you sure about this ? if `container_manager` is crio you are using `docker run` ?

Reply:
Ooh, you're right. I copied this from helm role. We should use crictl here. ctr is containerd-specific.